src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -197,15 +200,40 @@ public IRecordingDescriptor startRecording(
                     if (fixedDuration != null) {
                         Long delay =
                                 Long.valueOf(fixedDuration.toString().replaceAll("[^0-9]", ""));
-
                         scheduleRecordingTasks(
                                 recordingName, delay, connectionDescriptor, archiveOnStop);
                     }
-
                     return desc;
                 });
     }
 
+    private boolean shouldRestartRecording(String replace, String restart) {
+        if (replace != null) {
+            switch (replace) {
+                case "always":
+                    return true;
+                case "stopped":
+                    // Check if the recording is stopped
+                    boolean isStopped = isRecordingStopped(replace);
+                    return isStopped;
+                case "never":
+                    return false;
+            }
+        }
+        // Handle restart parameter (deprecated)
+        if (restart != null) {
```

Comment:
```java
if (restart != null) {
            boolean isStopped = isRecordingStopped(restart);
            return isStopped;
        }
        // if neither restart nor replace is specified, default to never
        return false;
```

This could just be `return restart != null && restart.equals("true");`